MonoGame.Framework/Game.cs

```diff
@@ -303,10 +303,12 @@ internal bool Initialized
 
         #region Events
 
+        private readonly ExitingEventArgs _exitingEventArgs = new ExitingEventArgs();
+
         public event EventHandler<EventArgs> Activated;
         public event EventHandler<EventArgs> Deactivated;
         public event EventHandler<EventArgs> Disposed;
-        public event EventHandler<EventArgs> Exiting;
+        public event EventHandler<ExitingEventArgs> Exiting;

```

Comment:
My concern here is that some MG titles are still using XNA on Windows.

This makes it inconvenient for them to work around this change.

IMO we should keep this as `EventArgs` and make it the responsibility of the implementer of `OnExiting` to cast to `ExitingEventArgs` if they want to use that feature.

I would use this opportunity to add documentation here to this event.

Reply:
This does not break anything, I assume you are worried about the following, but the event params do get autocast and it does compile:

```cs
// Inside MonoGame:
public class ExitingEventArgs() : EventArgs;
public static event EventHandler<ExitingEventArgs> Exiting;

// Inside the XNA based game:
public static void Main()
{
    Exiting += MyCustomExitEvent;
}

private static void MyCustomExitEvent(object sender, EventArgs args) { }
```

Reply:
Ahh... i had not thought of that @harry-cpp .

Reply:
Note that this does break user code that overrides OnExiting in a Game implementation because the signature changed.